lib/src/builder.dart

```diff
@@ -73,15 +93,24 @@ class PostgrestBuilder {
       );
     } else {
       dynamic body;
-      try {
-        body = json.decode(response.body);
-      } on FormatException catch (_) {
-        body = response.body;
+      int count;
+      if (response.request.method != 'HEAD') {
+        try {
+          body = json.decode(response.body);
+        } on FormatException catch (_) {
+          body = response.body;
+        }
       }
 
+      final contentRange = response.headers['content-range'];
+      count = contentRange.split('/')[1] == '*'
```

Comment:
we should have guard code here to prevent:
- null `contentRange`
- out of bound `contentRange.split('/')[1]`

Reply:
Thank you for spotting this. Let me come up with something.